libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2132,55 +2118,93 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 	// For positional calls only, store argument types
 	if (_functionCall.names().empty())
 	{
-		shared_ptr<TypePointers> const argumentTypes = make_shared<TypePointers>();
+		shared_ptr<TypePointers> argumentTypes = make_shared<TypePointers>();
 		for (ASTPointer<Expression const> const& argument: arguments)
 			argumentTypes->push_back(type(*argument));
 		_functionCall.expression().annotation().argumentTypes = move(argumentTypes);
 	}
 
 	_functionCall.expression().accept(*this);
+
 	TypePointer const expressionType = type(_functionCall.expression());
+	TypePointer actualType;
 
+	// Determine type for type conversion or struct construction expressions
 	if (auto const* const typeType = dynamic_cast<TypeType const*>(expressionType.get()))
+		actualType = typeType->actualType();
+
+	// Determine annotation kind for this FunctionCall node
+	const FunctionCallKind annoKind =
```

Comment:
The fact that this will be stored in the AST annotations later is secondary, I would say. So what about `functionCallKind` or `callKind` as name?

Reply:
Done.